lib/appraisal/matrix/rubygems_helper.rb

```diff
@@ -6,20 +6,43 @@
 module Appraisal
   module Matrix
     module RubygemsHelper
-      def versions_to_test(gem_name, minimum_version)
+      def versions_to_test(gem_name, minimum_version, maximum_version, step)
         # Generate a set to store the versions to test against 
         versions_to_test = Set.new
 
         # Load versions from rubygems api
         URI.parse("https://rubygems.org/api/v1/versions/#{gem_name}.json").open do |raw_version_data|
           JSON.parse(raw_version_data.read).each do |version_data|
             version = Gem::Version.new(version_data['number'])
-            versions_to_test << version.segments[0..1].join('.') if version >= minimum_version && !version.prerelease?
+            versions_to_test << version_segments(version, step).join('.') if include_version?(version, minimum_version, maximum_version)
           end
         end
 
         versions_to_test
       end
+
+      private
+
+      def version_segments(version, step)
+        case step
+        when :major
+          version.segments[0..0]
+        when :minor
+          version.segments[0..1]
+        when :patch
+          version.segments[0..2]
+        else
+          raise "Unsupported requested version step: #{step}"
+        end
+      end
+
+      def include_version?(version, minimum_version, maximum_version)
+        if maximum_version
+          !version.prerelease? && version >= minimum_version && version < maximum_version
+        else
+          !version.prerelease? && version >= minimum_version
+        end
```

Comment:
This can be simplified
```suggestion
        !version.prerelease? && version >= minimum_version && (!maximum_version || version < maximum_version)
```

Reply:
I think this is more readable in the if/else form. Do you feel strongly about this request?

Reply:
The point is to remove the duplicate `!version.prerelease? && version >= minimum_version`